File: Classes/core/TransportKit/STKMiniInstruction.m
//
//  STKMiniInstruction.m
//  TripGo
//
//  Created by Adrian Schoenig on 8/05/2015.
//  Copyright (c) 2015 SkedGo Pty Ltd. All rights reserved.
//

#import "STKMiniInstruction.h"

@implementation STKMiniInstruction

+ (nullable STKMiniInstruction *)miniInstructionForDictionary:(nullable NSDictionary *)dictionary
{
  if (!dictionary) {
    return nil;
  }
  
  STKMiniInstruction *mini  = [[self alloc] init];
  mini.instruction          = dictionary[@"instruction"];
  mini.mainValue            = dictionary[@"mainValue"];
  mini.detail               = dictionary[@"description"];
  return mini;
}

#pragma mark - NSCoding Protocol

- (void)encodeWithCoder:(NSCoder *)encoder
{
  [encoder encodeObject:_instruction forKey:@"instruction"];
  [encoder encodeObject:_mainValue forKey:@"mainValue"];
  [encoder encodeObject:_detail forKey:@"description"];
}

- (id)initWithCoder:(NSCoder *)decoder
{
  self = [super init];
  if (self) {
    self.instruction  = [decoder decodeObjectForKey:@"instruction"];
    self.mainValue    = [decoder decodeObjectForKey:@"mainValue"];
    self.detail       = [decoder decodeObjectForKey:@"description"];
  }
  return self;
}

@end

File: TripKit/Classes/core/TransportKit/STKMiniInstruction.swift
//
//  STKMiniInstruction.swift
//  TripKit
//
//  Created by Adrian Schönig on 05.04.18.
//  Copyright © 2018 SkedGo. All rights reserved.
//

import Foundation

public struct STKMiniInstruction: Codable, Equatable {
  public let instruction: String
  public let mainValue: String?
  public let detail: String?
  
  private enum CodingKeys: String, CodingKey {
    case instruction
    case mainValue
    case detail = "description"
  }
  
  public static func instruction(for json: [String: Any]?) -> STKMiniInstruction? {
    guard let json = json else { return nil }
    let decoder = JSONDecoder()
    return try? decoder.decode(STKMiniInstruction.self, withJSONObject: json)
  }

}
